analyses/mwi/dryspells_trigger/16_review_jan_feb_pred_threshold.md

```diff
@@ -0,0 +1,494 @@
+config=Config()
+parameters = config.parameters(country_iso3)
+
+country_data_raw_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR, config.RAW_DIR,country_iso3)
+country_data_processed_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,config.PROCESSED_DIR,country_iso3)
+country_data_exploration_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,"exploration",country_iso3)
+chirps_country_data_exploration_dir= os.path.join(config.DATA_DIR,config.PUBLIC_DIR, "exploration", country_iso3,'chirps')
+
+chirps_monthly_mwi_path=os.path.join(chirps_country_data_exploration_dir,"chirps_mwi_monthly.nc")
+
+monthly_precip_exploration_dir=os.path.join(country_data_exploration_dir,"dryspells", f"v{parameters['version']}", "monthly_precipitation")
+dry_spells_processed_dir = os.path.join(country_data_processed_dir, "dry_spells", f"v{parameters['version']}")
+
+plots_dir=os.path.join(country_data_processed_dir,"plots","dry_spells")
+plots_seasonal_dir=os.path.join(plots_dir,"seasonal")
+
+adm2_bound_path=os.path.join(country_data_raw_dir,config.SHAPEFILE_DIR,parameters["path_admin2_shp"])
+all_dry_spells_list_path=os.path.join(dry_spells_processed_dir,"full_list_dry_spells_2000_2021.csv")
+monthly_precip_path=os.path.join(country_data_processed_dir,"chirps","chirps_monthly_total_precipitation_admin1.csv")
+```
+
+
+```python
+#using the mean value of the admin
+if use_unrounded_area_coords:
+    plots_seasonal_dir = Path(plots_seasonal_dir) / "unrounded-coords"
+aggr_meth = "mean_ADM1_PCODE"
+```
+
+
+```python
+#set plot colors
+ds_color='#F2645A'
+no_ds_color='#CCE5F9'
+no_ds_color_dark='#66B0EC'
+```
+
+
+```python
+def det_rate(tp,fn,epsilon):
+    return tp/(tp+fn+epsilon)*100
+def tn_rate(tn,fp,epsilon):
+    return tn/(tn+fp+epsilon)*100
+def miss_rate(fn,tp,epsilon):
+    return fn/(tp+fn+epsilon)*100
+def false_alarm_rate(fp,tp,epsilon):
+    return fp/(tp+fp+epsilon)*100
+
+def compute_miss_false_leadtime(df,target_var,predict_var):
+    df_pr=pd.DataFrame(list(df.leadtime.unique()),columns=["leadtime"]).set_index('leadtime')
+    #TODO: also account for different adm1's
+    a="Southern"
+    for i, m in enumerate(df.sort_values(by="leadtime").leadtime.unique()):
+        y_target =    df.loc[df.leadtime==m,target_var]
+        y_predicted = df.loc[df.leadtime==m,predict_var]
+        
+        cm = confusion_matrix(y_target=y_target, 
+                              y_predicted=y_predicted)
+
+        tn,fp,fn,tp=cm.flatten()
+        df_pr.loc[m,["month_ds"]] = det_rate(tp,fn,epsilon)
+        df_pr.loc[m,["month_no_ds"]]= tn_rate(tn,fp,epsilon)
+        df_pr.loc[m,["month_miss_rate"]]= miss_rate(fn,tp,epsilon)
+        df_pr.loc[m,["month_false_alarm_rate"]]= false_alarm_rate(fp,tp,epsilon)
+        df_pr.loc[m,["tn","tp","fp","fn"]]=tn,tp,fp,fn
+    df_pr=df_pr.reset_index()
+    return df_pr
+```
+
+
+```python
+def compute_confusionmatrix_leadtime(df,target_var,predict_var, ylabel,xlabel,colp_num=3,title=None,figsize=(15,8)):
```

Comment:
It seems this function and the next two are not used so you could delete them. But if you think you might use them in further exploration, you can leave them ofc.